kats/src/main/kotlin/kats/Option.kt

```diff
@@ -24,6 +24,11 @@ package kats
  */
 sealed class Option<out A> {
 
+    companion object {
+
+        inline operator fun <A> invoke(f: () -> A): Option<A> = f().let { if (it == null) None else Some(it) }
```

Comment:
Even though Scala does that. Additionally if you want this behaviour we should add a factory method `fromNullable` which does what you have implemented here